src/Aspire.Hosting.Azure.Sql/AzureSqlServerResource.cs

```diff
@@ -108,11 +115,76 @@ public override ProvisionableResource AddAsExistingResource(AzureResourceInfrast
     public override void AddRoleAssignments(IAddRoleAssignmentsContext roleAssignmentContext)
     {
         var infra = roleAssignmentContext.Infrastructure;
-        var postgres = (SqlServer)AddAsExistingResource(infra);
+        var sqlserver = (SqlServer)AddAsExistingResource(infra);
 
-        var principalId = roleAssignmentContext.PrincipalId;
+        // The name of the user assigned identity for the container app
         var principalName = roleAssignmentContext.PrincipalName;
+        var principalId = roleAssignmentContext.PrincipalId;
```

Comment:
I think you need to extend https://github.com/dotnet/aspire/blob/935f06b15acaa8068375d5507423aacdaaf1de52/src/Aspire.Hosting.Azure/IAddRoleAssignmentsContext.cs#L32 to expose the client id.

cc @eerhardt

Reply:
We actually need the `principalId` here

Reply:
Uses ClientId now